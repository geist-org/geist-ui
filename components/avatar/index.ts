import Avatar from './avatar'
import AvatarGroup from './avatar-group'

export type AvatarComponentType = typeof Avatar & {
  Group: typeof AvatarGroup
}
;(Avatar as AvatarComponentType).Group = AvatarGroup

export type { AvatarProps } from './avatar'
export type { AvatarGroupProps } from './avatar-group'

export default Avatar as AvatarComponentType
