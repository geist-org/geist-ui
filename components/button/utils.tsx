import React, { ReactNode } from 'react'
import ButtonIcon from './button-icon'
import { ButtonProps } from './button'
import { ButtonGroupConfig } from '../button-group/button-group-context'
import useClasses from '../use-classes'

export const getButtonChildrenWithIcon = (
  auto: boolean,
  children: ReactNode,
  icons: {
    icon?: React.ReactNode
    iconRight?: React.ReactNode
  },
) => {
  const { icon, iconRight } = icons
  const hasIcon = icon || iconRight
  const isRight = Boolean(iconRight)
  const paddingForAutoMode = auto
    ? `calc(var(--geist-ui-button-height) / 2 + var(--geist-ui-button-icon-padding) * .5)`
    : 0
  const classes = useClasses('text', isRight ? 'right' : 'left')

  if (!hasIcon) return <div className="text">{children}</div>
  if (React.Children.count(children) === 0) {
    return (
      <ButtonIcon isRight={isRight} isSingle>
        {hasIcon}
      </ButtonIcon>
    )
  }
  return (
    <>
      <ButtonIcon isRight={isRight}>{hasIcon}</ButtonIcon>
      <div className={classes}>
        {children}
        <style jsx>{`
          .left {
            padding-left: ${paddingForAutoMode};
          }
          .right {
            padding-right: ${paddingForAutoMode};
          }
        `}</style>
      </div>
    </>
  )
}

export const filterPropsWithGroup = <T extends React.PropsWithChildren<ButtonProps>>(
  props: T,
  config: ButtonGroupConfig,
): T => {
  if (!config.isButtonGroup) return props
  return {
    ...props,
    auto: true,
    shadow: false,
    ghost: config.ghost || props.ghost,
    type: config.type || props.type,
    disabled: config.disabled || props.disabled,
  }
}
