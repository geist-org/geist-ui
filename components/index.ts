// @ts-ignore
/// <reference types="styled-jsx" />

export { default as AutoComplete } from './auto-complete'
export type { AutoCompleteProps } from './auto-complete'

export { default as Avatar } from './avatar'
export type { AvatarProps, AvatarGroupProps } from './avatar'

export { default as Badge } from './badge'
export type { BadgeProps, BadgeAnchorProps } from './badge'

export { default as Breadcrumbs } from './breadcrumbs'
export type {
  BreadcrumbsProps,
  BreadcrumbsItemProps,
  BreadcrumbsSeparatorProps,
} from './breadcrumbs'

export { default as Button } from './button'
export type { ButtonProps } from './button'

export { default as ButtonDropdown } from './button-dropdown'
export type { ButtonDropdownProps, ButtonDropdownItemProps } from './button-dropdown'

export { default as ButtonGroup } from './button-group'
export type { ButtonGroupProps } from './button-group'

export { default as Capacity } from './capacity'
export type { CapacityProps } from './capacity'

export { default as Card } from './card'
export type { CardProps, CardContentProps, CardFooterProps } from './card'

export { default as Checkbox } from './checkbox'
export type { CheckboxProps, CheckboxGroupProps } from './checkbox'

export { default as Code } from './code'
export type { CodeProps } from './code'

export { default as Collapse } from './collapse'
export type { CollapseProps, CollapseGroupProps } from './collapse'

export { default as Description } from './description'
export type { DescriptionProps } from './description'

export { default as Display } from './display'
export type { DisplayProps } from './display'

export { default as Divider } from './divider'
export type { DividerProps } from './divider'

export { default as Dot } from './dot'
export type { DotProps } from './dot'

export { default as Drawer } from './drawer'
export type { DrawerProps } from './drawer'

export { default as Fieldset } from './fieldset'
export type {
  FieldsetProps,
  FieldsetTitleProps,
  FieldsetSubtitleProps,
  FieldsetGroupProps,
  FieldsetFooterProps,
  FieldsetContentProps,
} from './fieldset'

export { default as GeistProvider } from './geist-provider'
export type { GeistProviderProps } from './geist-provider'

export { default as Grid } from './grid'
export type { GridProps, GridContainerProps } from './grid'

export { default as Image } from './image'
export type { ImageProps, ImageBrowserProps } from './image'

export { default as Input } from './input'
export type { InputProps, InputPasswordProps } from './input'

export { default as Keyboard } from './keyboard'
export type { KeyboardProps } from './keyboard'

export { default as Link } from './link'
export type { LinkProps } from './link'

export { default as Loading } from './loading'
export type { LoadingProps } from './loading'

export { default as Modal } from './modal'
export type {
  ModalProps,
  ModalTitleProps,
  ModalSubtitleProps,
  ModalContentProps,
  ModalActionProps,
} from './modal'

export { default as Note } from './note'
export type { NoteProps } from './note'

export { default as Page } from './page'
export type {
  PageProps,
  PageHeaderProps,
  PageContentProps,
  PageFooterProps,
} from './page'

export { default as Pagination } from './pagination'
export type {
  PaginationProps,
  PaginationNextProps,
  PaginationPreviousProps,
} from './pagination'

export { default as Popover } from './popover'
export type { PopoverProps, PopoverItemProps } from './popover'

export { default as Progress } from './progress'
export type { ProgressProps } from './progress'

export { default as Radio } from './radio'
export type { RadioProps, RadioGroupProps, RadioDescriptionProps } from './radio'

export { default as Rating } from './rating'
export type { RatingProps } from './rating'

export { default as Select } from './select'
export type { SelectProps, SelectOptionProps } from './select'

export { default as Slider } from './slider'
export type { SliderProps } from './slider'

export { default as Snippet } from './snippet'
export type { SnippetProps } from './snippet'

export { default as Spacer } from './spacer'
export type { SpacerProps } from './spacer'

export { default as Spinner } from './spinner'
export type { SpinnerProps } from './spinner'

export { default as Table } from './table'
export type { TableProps, TableColumnProps } from './table'

export { default as Tabs } from './tabs'
export type { TabsProps } from './tabs'

export { default as Tag } from './tag'
export type { TagProps } from './tag'

export { default as Text } from './text'
export type { TextProps } from './text'

export { default as Textarea } from './textarea'
export type { TextareaProps } from './textarea'

export { default as Themes } from './themes'
export type { GeistUIThemes, GeistUserTheme } from './themes'

export { default as Toggle } from './toggle'
export type { ToggleProps } from './toggle'

export { default as Tooltip } from './tooltip'
export type { TooltipProps } from './tooltip'

export { default as Tree } from './tree'
export type { TreeProps } from './tree'

export { default as useAllThemes } from './use-all-themes'
export type { AllThemesConfig } from './use-all-themes'

export { default as useToasts } from './use-toasts'
export type { Toast, ToastInput, ToastAction, ToastLayout } from './use-toasts'

export { default as User } from './user'
export type { UserProps } from './user'

export { default as useBodyScroll } from './use-body-scroll'
export type { BodyScrollOptions } from './use-body-scroll'

export { default as useClipboard } from './use-clipboard'
export type { UseClipboardOptions } from './use-clipboard'

export { default as useMediaQuery } from './use-media-query'
export type { ResponsiveOptions, ResponsiveBreakpoint } from './use-media-query'

export { default as useKeyboard, KeyMod, KeyCode } from './use-keyboard'
export type { KeyboardOptions, UseKeyboardHandler } from './use-keyboard'

export { default as useInput } from './use-input'
export { default as useModal } from './use-modal'
export { default as useTabs } from './use-tabs'
export { default as useClickAway } from './use-click-away'
export { default as useCurrentState } from './use-current-state'
export { default as CssBaseline } from './css-baseline'
export { default as useTheme } from './use-theme'
export { default as useClasses } from './use-classes'
export { default as useScale } from './use-scale'
export { withScale, ScalePropKeys, ScaleContext } from './use-scale'
export type {
  ScaleProps,
  ScaleConfig,
  GetScalePropsFunction,
  GetAllScalePropsFunction,
} from './use-scale'

export { useRect } from './utils/layouts'
export type { ReactiveDomReact } from './utils/layouts'
export { default as Highlight } from './shared/highlight'
export type { HighlightProps } from './shared/highlight'
