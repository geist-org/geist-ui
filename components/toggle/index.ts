import Toggle from './toggle'

export type {
  ToggleProps,
  ToggleSize,
  ToggleEvent,
  ToggleEventTarget,
  ToggleTypes,
} from './toggle'
export default Toggle
