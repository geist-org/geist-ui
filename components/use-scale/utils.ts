import {
  GetAllScalePropsFunction,
  GetScalePropsFunction,
  ScaleProps,
  ScalePropKeys,
} from './scale-context'

export const generateGetScaleProps = <P>(
  props: P & ScaleProps,
): GetScalePropsFunction => {
  const getScaleProps: GetScalePropsFunction = keyOrKeys => {
    if (!Array.isArray(keyOrKeys)) return props[keyOrKeys as keyof ScaleProps]
    let value = undefined
    for (const key of keyOrKeys) {
      const currentValue = props[key]
      if (typeof currentValue !== 'undefined') {
        value = currentValue
      }
    }
    return value
  }
  return getScaleProps
}

export const generateGetAllScaleProps = <P>(
  props: P & ScaleProps,
): GetAllScalePropsFunction => {
  const getAllScaleProps: GetAllScalePropsFunction = () => {
    let scaleProps: ScaleProps = {}
    for (const key of ScalePropKeys) {
      const value = props[key as keyof ScaleProps]
      if (typeof value !== 'undefined') {
        scaleProps[key as keyof ScaleProps] = value as any
      }
    }
    return scaleProps
  }
  return getAllScaleProps
}
