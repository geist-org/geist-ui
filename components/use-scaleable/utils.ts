import { ScaleableProps } from './scaleable-context'

export type ScaleablePropsAndInvalid = keyof ScaleableProps | 'size'

export const ScaleablePropKeys: Array<ScaleablePropsAndInvalid> = [
  'paddingLeft',
  'pl',
  'paddingRight',
  'pr',
  'paddingTop',
  'pt',
  'paddingBottom',
  'pb',
  'marginTop',
  'mt',
  'marginRight',
  'mr',
  'marginBottom',
  'mb',
  'marginLeft',
  'ml',
  'px',
  'py',
  'mx',
  'my',
  'width',
  'height',
  'font',
  'unit',
  'scale',
  'size',
]

export const filterScaleableProps = <T extends Record<any, any>>(props: T) => {
  const keys = Object.keys(props).filter(key => key !== '')
  const nextProps: any = {}
  for (const key of keys) {
    if (!(ScaleablePropKeys as string[]).includes(key)) {
      nextProps[key] = props[key]
    }
  }
  return nextProps as Omit<T, ScaleablePropsAndInvalid>
}
