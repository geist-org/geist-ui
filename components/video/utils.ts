export const formatTime = (seconds: number) => {
  const hh = Math.floor(seconds / 3600)
  const mm = Math.floor(seconds / 60) % 60
  const ss = Math.floor(seconds) % 60
  const pad = (num: number) => (num < 10 ? '0' + num : num)
  return `${hh ? pad(hh) + ':' : ''}${pad(mm)}:${pad(ss)}`
}

export const isFullscreenEnabled = () => {
  const elem = document as any
  return (
    elem.fullscreenEnabled ||
    elem.webkitFullscreenEnabled ||
    elem.mozFullScreenEnabled ||
    elem.msFullscreenEnabled
  )
}

export const openFullscreen = (element: HTMLElement) => {
  const elem = element as any
  if (elem.requestFullscreen) {
    elem.requestFullscreen()
  } else if (elem.mozRequestFullScreen) {
    elem.mozRequestFullScreen()
  } else if (elem.webkitRequestFullscreen) {
    elem.webkitRequestFullscreen()
  } else if (elem.msRequestFullscreen) {
    elem.msRequestFullscreen()
  }
}

export const closeFullscreen = () => {
  const elem = document as any
  if (elem.exitFullscreen) {
    elem.exitFullscreen()
  } else if (elem.mozCancelFullScreen) {
    elem.mozCancelFullScreen()
  } else if (elem.webkitExitFullscreen) {
    elem.webkitExitFullscreen()
  } else if (elem.msExitFullscreen) {
    elem.msExitFullscreen()
  }
}

export const onFullscreenChange = (func: (state: boolean) => void) => {
  const doc = document as any
  doc.addEventListener('fullscreenchange', () => func(!!doc.fullscreen), true)
  doc.addEventListener('mozfullscreenchange', () => func(!!doc.mozFullScreen), true)
  doc.addEventListener('webkitfullscreenchange', () => func(!!doc.webkitIsFullScreen), true)
  doc.addEventListener('msfullscreenchange', () => func(!!doc.msFullscreenElement), true)
}

export const isSafari = () =>
  navigator.userAgent.includes('Safari') && !navigator.userAgent.includes('Chrome')
