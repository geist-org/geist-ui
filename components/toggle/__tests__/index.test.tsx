import React from 'react'
import { mount, ReactWrapper } from 'enzyme'
import { Toggle } from 'components'
import { nativeEvent, updateWrapper } from 'tests/utils'

const expectToggleIsChecked = (wrapper: ReactWrapper) => {
  expect(wrapper.find('.checked').length).not.toBe(0)
}

const expectToggleIsUnChecked = (wrapper: ReactWrapper) => {
  expect(wrapper.find('.checked').length).toBe(0)
}

describe('Toggle', () => {
  it('should render correctly', () => {
    const wrapper = mount(<Toggle />)
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should work with different sizes', () => {
    const wrapper = mount(
      <div>
        <Toggle scale={1} />
        <Toggle scale={0.2} />
        <Toggle scale={2} />
        <Toggle unit="10px" />
      </div>,
    )
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should work with different status', () => {
    const wrapper = mount(
      <div>
        <Toggle type="secondary" />
        <Toggle type="success" />
        <Toggle type="warning" />
        <Toggle type="error" />
      </div>,
    )
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should set toggle follow checked prop', async () => {
    const wrapper = mount(<Toggle initialChecked={true} />)
    expectToggleIsChecked(wrapper)

    wrapper.setProps({ checked: false })
    await updateWrapper(wrapper)
    expectToggleIsUnChecked(wrapper)

    wrapper.setProps({ checked: true })
    await updateWrapper(wrapper)
    expectToggleIsChecked(wrapper)
  })

  it('should trigger events when toggle changed', async () => {
    let checked = false
    const changeHandler = jest.fn().mockImplementation(e => (checked = e.target.checked))
    const wrapper = mount(<Toggle onChange={changeHandler} />)

    wrapper.find('input').simulate('change', {
      ...nativeEvent,
      target: { checked: true },
    })
    await updateWrapper(wrapper)
    expectToggleIsChecked(wrapper)

    expect(changeHandler).toHaveBeenCalled()
    expect(checked).toEqual(true)
  })

  it('should ignore events when toggle disabled', async () => {
    const changeHandler = jest.fn()
    const wrapper = mount(<Toggle onChange={changeHandler} disabled />)

    wrapper.find('input').simulate('change', {
      ...nativeEvent,
      target: { checked: true },
    })
    await updateWrapper(wrapper)
    expectToggleIsUnChecked(wrapper)

    expect(changeHandler).not.toHaveBeenCalled()
  })
})
