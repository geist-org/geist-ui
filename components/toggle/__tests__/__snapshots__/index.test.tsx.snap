// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle should render correctly 1`] = `
"<label class=\\"\\"><input type=\\"checkbox\\"><div class=\\"toggle\\"><span class=\\"inner\\"></span></div><style>
        label {
          -webkit-tap-highlight-color: transparent;
          display: inline-block;
          vertical-align: middle;
          white-space: nowrap;
          user-select: none;
          position: relative;
          cursor: pointer;
          --toggle-font-size: calc(1 * 16px);
          --toggle-height: calc(0.875 * 16px);
          width: calc(1.75 * 16px);
          height: var(--toggle-height);
          padding: calc(0.1875 * 16px) 0 calc(0.1875 * 16px)
            0;
          margin: 0 0 0 0;
        }

        input {
          overflow: hidden;
          visibility: hidden;
          height: 0;
          opacity: 0;
          width: 0;
          position: absolute;
          background-color: transparent;
          z-index: -1;
        }

        .toggle {
          height: var(--toggle-height);
          width: 100%;
          border-radius: var(--toggle-height);
          transition-delay: 0.12s;
          transition-duration: 0.2s;
          transition-property: background, border;
          transition-timing-function: cubic-bezier(0, 0, 0.2, 1);
          position: relative;
          border: 1px solid transparent;
          background-color: #eaeaea;
          padding: 0;
        }

        .inner {
          width: calc(var(--toggle-height) - 2px);
          height: calc(var(--toggle-height) - 2px);
          position: absolute;
          top: 50%;
          transform: translateY(-50%);
          left: 1px;
          box-shadow: rgba(0, 0, 0, 0.2) 0 1px 2px 0, rgba(0, 0, 0, 0.1) 0 1px 3px 0;
          transition: left 280ms cubic-bezier(0, 0, 0.2, 1);
          border-radius: 50%;
          background-color: #fff;
        }

        .disabled {
          border-color: #eaeaea;
          background-color: #fafafa;
        }

        .disabled > .inner {
          background-color: #eaeaea;
        }

        .disabled.checked {
          border-color: #888;
          background-color: #888;
        }

        .checked {
          background-color: #0070f3;
        }

        .checked > .inner {
          left: calc(100% - (var(--toggle-height) - 2px));
          box-shadow: none;
        }
      </style></label>"
`;

exports[`Toggle should work with different sizes 1`] = `
"<div><label class=\\"\\"><input type=\\"checkbox\\"><div class=\\"toggle\\"><span class=\\"inner\\"></span></div><style>
        label {
          -webkit-tap-highlight-color: transparent;
          display: inline-block;
          vertical-align: middle;
          white-space: nowrap;
          user-select: none;
          position: relative;
          cursor: pointer;
          --toggle-font-size: calc(1 * 16px);
          --toggle-height: calc(0.875 * 16px);
          width: calc(1.75 * 16px);
          height: var(--toggle-height);
          padding: calc(0.1875 * 16px) 0 calc(0.1875 * 16px)
            0;
          margin: 0 0 0 0;
        }

        input {
          overflow: hidden;
          visibility: hidden;
          height: 0;
          opacity: 0;
          width: 0;
          position: absolute;
          background-color: transparent;
          z-index: -1;
        }

        .toggle {
          height: var(--toggle-height);
          width: 100%;
          border-radius: var(--toggle-height);
          transition-delay: 0.12s;
          transition-duration: 0.2s;
          transition-property: background, border;
          transition-timing-function: cubic-bezier(0, 0, 0.2, 1);
          position: relative;
          border: 1px solid transparent;
          background-color: #eaeaea;
          padding: 0;
        }

        .inner {
          width: calc(var(--toggle-height) - 2px);
          height: calc(var(--toggle-height) - 2px);
          position: absolute;
          top: 50%;
          transform: translateY(-50%);
          left: 1px;
          box-shadow: rgba(0, 0, 0, 0.2) 0 1px 2px 0, rgba(0, 0, 0, 0.1) 0 1px 3px 0;
          transition: left 280ms cubic-bezier(0, 0, 0.2, 1);
          border-radius: 50%;
          background-color: #fff;
        }

        .disabled {
          border-color: #eaeaea;
          background-color: #fafafa;
        }

        .disabled > .inner {
          background-color: #eaeaea;
        }

        .disabled.checked {
          border-color: #888;
          background-color: #888;
        }

        .checked {
          background-color: #0070f3;
        }

        .checked > .inner {
          left: calc(100% - (var(--toggle-height) - 2px));
          box-shadow: none;
        }
      </style></label><label class=\\"\\"><input type=\\"checkbox\\"><div class=\\"toggle\\"><span class=\\"inner\\"></span></div><style>
        label {
          -webkit-tap-highlight-color: transparent;
          display: inline-block;
          vertical-align: middle;
          white-space: nowrap;
          user-select: none;
          position: relative;
          cursor: pointer;
          --toggle-font-size: calc(0.6 * 16px);
          --toggle-height: calc(0.525 * 16px);
          width: calc(1.05 * 16px);
          height: var(--toggle-height);
          padding: calc(0.11249999999999999 * 16px) 0 calc(0.11249999999999999 * 16px)
            0;
          margin: 0 0 0 0;
        }

        input {
          overflow: hidden;
          visibility: hidden;
          height: 0;
          opacity: 0;
          width: 0;
          position: absolute;
          background-color: transparent;
          z-index: -1;
        }

        .toggle {
          height: var(--toggle-height);
          width: 100%;
          border-radius: var(--toggle-height);
          transition-delay: 0.12s;
          transition-duration: 0.2s;
          transition-property: background, border;
          transition-timing-function: cubic-bezier(0, 0, 0.2, 1);
          position: relative;
          border: 1px solid transparent;
          background-color: #eaeaea;
          padding: 0;
        }

        .inner {
          width: calc(var(--toggle-height) - 2px);
          height: calc(var(--toggle-height) - 2px);
          position: absolute;
          top: 50%;
          transform: translateY(-50%);
          left: 1px;
          box-shadow: rgba(0, 0, 0, 0.2) 0 1px 2px 0, rgba(0, 0, 0, 0.1) 0 1px 3px 0;
          transition: left 280ms cubic-bezier(0, 0, 0.2, 1);
          border-radius: 50%;
          background-color: #fff;
        }

        .disabled {
          border-color: #eaeaea;
          background-color: #fafafa;
        }

        .disabled > .inner {
          background-color: #eaeaea;
        }

        .disabled.checked {
          border-color: #888;
          background-color: #888;
        }

        .checked {
          background-color: #0070f3;
        }

        .checked > .inner {
          left: calc(100% - (var(--toggle-height) - 2px));
          box-shadow: none;
        }
      </style></label><label class=\\"\\"><input type=\\"checkbox\\"><div class=\\"toggle\\"><span class=\\"inner\\"></span></div><style>
        label {
          -webkit-tap-highlight-color: transparent;
          display: inline-block;
          vertical-align: middle;
          white-space: nowrap;
          user-select: none;
          position: relative;
          cursor: pointer;
          --toggle-font-size: calc(1.5 * 16px);
          --toggle-height: calc(1.3125 * 16px);
          width: calc(2.625 * 16px);
          height: var(--toggle-height);
          padding: calc(0.28125 * 16px) 0 calc(0.28125 * 16px)
            0;
          margin: 0 0 0 0;
        }

        input {
          overflow: hidden;
          visibility: hidden;
          height: 0;
          opacity: 0;
          width: 0;
          position: absolute;
          background-color: transparent;
          z-index: -1;
        }

        .toggle {
          height: var(--toggle-height);
          width: 100%;
          border-radius: var(--toggle-height);
          transition-delay: 0.12s;
          transition-duration: 0.2s;
          transition-property: background, border;
          transition-timing-function: cubic-bezier(0, 0, 0.2, 1);
          position: relative;
          border: 1px solid transparent;
          background-color: #eaeaea;
          padding: 0;
        }

        .inner {
          width: calc(var(--toggle-height) - 2px);
          height: calc(var(--toggle-height) - 2px);
          position: absolute;
          top: 50%;
          transform: translateY(-50%);
          left: 1px;
          box-shadow: rgba(0, 0, 0, 0.2) 0 1px 2px 0, rgba(0, 0, 0, 0.1) 0 1px 3px 0;
          transition: left 280ms cubic-bezier(0, 0, 0.2, 1);
          border-radius: 50%;
          background-color: #fff;
        }

        .disabled {
          border-color: #eaeaea;
          background-color: #fafafa;
        }

        .disabled > .inner {
          background-color: #eaeaea;
        }

        .disabled.checked {
          border-color: #888;
          background-color: #888;
        }

        .checked {
          background-color: #0070f3;
        }

        .checked > .inner {
          left: calc(100% - (var(--toggle-height) - 2px));
          box-shadow: none;
        }
      </style></label><label class=\\"\\"><input type=\\"checkbox\\"><div class=\\"toggle\\"><span class=\\"inner\\"></span></div><style>
        label {
          -webkit-tap-highlight-color: transparent;
          display: inline-block;
          vertical-align: middle;
          white-space: nowrap;
          user-select: none;
          position: relative;
          cursor: pointer;
          --toggle-font-size: calc(1 * 10px);
          --toggle-height: calc(0.875 * 10px);
          width: calc(1.75 * 10px);
          height: var(--toggle-height);
          padding: calc(0.1875 * 10px) 0 calc(0.1875 * 10px)
            0;
          margin: 0 0 0 0;
        }

        input {
          overflow: hidden;
          visibility: hidden;
          height: 0;
          opacity: 0;
          width: 0;
          position: absolute;
          background-color: transparent;
          z-index: -1;
        }

        .toggle {
          height: var(--toggle-height);
          width: 100%;
          border-radius: var(--toggle-height);
          transition-delay: 0.12s;
          transition-duration: 0.2s;
          transition-property: background, border;
          transition-timing-function: cubic-bezier(0, 0, 0.2, 1);
          position: relative;
          border: 1px solid transparent;
          background-color: #eaeaea;
          padding: 0;
        }

        .inner {
          width: calc(var(--toggle-height) - 2px);
          height: calc(var(--toggle-height) - 2px);
          position: absolute;
          top: 50%;
          transform: translateY(-50%);
          left: 1px;
          box-shadow: rgba(0, 0, 0, 0.2) 0 1px 2px 0, rgba(0, 0, 0, 0.1) 0 1px 3px 0;
          transition: left 280ms cubic-bezier(0, 0, 0.2, 1);
          border-radius: 50%;
          background-color: #fff;
        }

        .disabled {
          border-color: #eaeaea;
          background-color: #fafafa;
        }

        .disabled > .inner {
          background-color: #eaeaea;
        }

        .disabled.checked {
          border-color: #888;
          background-color: #888;
        }

        .checked {
          background-color: #0070f3;
        }

        .checked > .inner {
          left: calc(100% - (var(--toggle-height) - 2px));
          box-shadow: none;
        }
      </style></label></div>"
`;

exports[`Toggle should work with different status 1`] = `
"<div><label class=\\"\\"><input type=\\"checkbox\\"><div class=\\"toggle\\"><span class=\\"inner\\"></span></div><style>
        label {
          -webkit-tap-highlight-color: transparent;
          display: inline-block;
          vertical-align: middle;
          white-space: nowrap;
          user-select: none;
          position: relative;
          cursor: pointer;
          --toggle-font-size: calc(1 * 16px);
          --toggle-height: calc(0.875 * 16px);
          width: calc(1.75 * 16px);
          height: var(--toggle-height);
          padding: calc(0.1875 * 16px) 0 calc(0.1875 * 16px)
            0;
          margin: 0 0 0 0;
        }

        input {
          overflow: hidden;
          visibility: hidden;
          height: 0;
          opacity: 0;
          width: 0;
          position: absolute;
          background-color: transparent;
          z-index: -1;
        }

        .toggle {
          height: var(--toggle-height);
          width: 100%;
          border-radius: var(--toggle-height);
          transition-delay: 0.12s;
          transition-duration: 0.2s;
          transition-property: background, border;
          transition-timing-function: cubic-bezier(0, 0, 0.2, 1);
          position: relative;
          border: 1px solid transparent;
          background-color: #eaeaea;
          padding: 0;
        }

        .inner {
          width: calc(var(--toggle-height) - 2px);
          height: calc(var(--toggle-height) - 2px);
          position: absolute;
          top: 50%;
          transform: translateY(-50%);
          left: 1px;
          box-shadow: rgba(0, 0, 0, 0.2) 0 1px 2px 0, rgba(0, 0, 0, 0.1) 0 1px 3px 0;
          transition: left 280ms cubic-bezier(0, 0, 0.2, 1);
          border-radius: 50%;
          background-color: #fff;
        }

        .disabled {
          border-color: #eaeaea;
          background-color: #fafafa;
        }

        .disabled > .inner {
          background-color: #eaeaea;
        }

        .disabled.checked {
          border-color: #888;
          background-color: #888;
        }

        .checked {
          background-color: #666;
        }

        .checked > .inner {
          left: calc(100% - (var(--toggle-height) - 2px));
          box-shadow: none;
        }
      </style></label><label class=\\"\\"><input type=\\"checkbox\\"><div class=\\"toggle\\"><span class=\\"inner\\"></span></div><style>
        label {
          -webkit-tap-highlight-color: transparent;
          display: inline-block;
          vertical-align: middle;
          white-space: nowrap;
          user-select: none;
          position: relative;
          cursor: pointer;
          --toggle-font-size: calc(1 * 16px);
          --toggle-height: calc(0.875 * 16px);
          width: calc(1.75 * 16px);
          height: var(--toggle-height);
          padding: calc(0.1875 * 16px) 0 calc(0.1875 * 16px)
            0;
          margin: 0 0 0 0;
        }

        input {
          overflow: hidden;
          visibility: hidden;
          height: 0;
          opacity: 0;
          width: 0;
          position: absolute;
          background-color: transparent;
          z-index: -1;
        }

        .toggle {
          height: var(--toggle-height);
          width: 100%;
          border-radius: var(--toggle-height);
          transition-delay: 0.12s;
          transition-duration: 0.2s;
          transition-property: background, border;
          transition-timing-function: cubic-bezier(0, 0, 0.2, 1);
          position: relative;
          border: 1px solid transparent;
          background-color: #eaeaea;
          padding: 0;
        }

        .inner {
          width: calc(var(--toggle-height) - 2px);
          height: calc(var(--toggle-height) - 2px);
          position: absolute;
          top: 50%;
          transform: translateY(-50%);
          left: 1px;
          box-shadow: rgba(0, 0, 0, 0.2) 0 1px 2px 0, rgba(0, 0, 0, 0.1) 0 1px 3px 0;
          transition: left 280ms cubic-bezier(0, 0, 0.2, 1);
          border-radius: 50%;
          background-color: #fff;
        }

        .disabled {
          border-color: #eaeaea;
          background-color: #fafafa;
        }

        .disabled > .inner {
          background-color: #eaeaea;
        }

        .disabled.checked {
          border-color: #888;
          background-color: #888;
        }

        .checked {
          background-color: #0070f3;
        }

        .checked > .inner {
          left: calc(100% - (var(--toggle-height) - 2px));
          box-shadow: none;
        }
      </style></label><label class=\\"\\"><input type=\\"checkbox\\"><div class=\\"toggle\\"><span class=\\"inner\\"></span></div><style>
        label {
          -webkit-tap-highlight-color: transparent;
          display: inline-block;
          vertical-align: middle;
          white-space: nowrap;
          user-select: none;
          position: relative;
          cursor: pointer;
          --toggle-font-size: calc(1 * 16px);
          --toggle-height: calc(0.875 * 16px);
          width: calc(1.75 * 16px);
          height: var(--toggle-height);
          padding: calc(0.1875 * 16px) 0 calc(0.1875 * 16px)
            0;
          margin: 0 0 0 0;
        }

        input {
          overflow: hidden;
          visibility: hidden;
          height: 0;
          opacity: 0;
          width: 0;
          position: absolute;
          background-color: transparent;
          z-index: -1;
        }

        .toggle {
          height: var(--toggle-height);
          width: 100%;
          border-radius: var(--toggle-height);
          transition-delay: 0.12s;
          transition-duration: 0.2s;
          transition-property: background, border;
          transition-timing-function: cubic-bezier(0, 0, 0.2, 1);
          position: relative;
          border: 1px solid transparent;
          background-color: #eaeaea;
          padding: 0;
        }

        .inner {
          width: calc(var(--toggle-height) - 2px);
          height: calc(var(--toggle-height) - 2px);
          position: absolute;
          top: 50%;
          transform: translateY(-50%);
          left: 1px;
          box-shadow: rgba(0, 0, 0, 0.2) 0 1px 2px 0, rgba(0, 0, 0, 0.1) 0 1px 3px 0;
          transition: left 280ms cubic-bezier(0, 0, 0.2, 1);
          border-radius: 50%;
          background-color: #fff;
        }

        .disabled {
          border-color: #eaeaea;
          background-color: #fafafa;
        }

        .disabled > .inner {
          background-color: #eaeaea;
        }

        .disabled.checked {
          border-color: #888;
          background-color: #888;
        }

        .checked {
          background-color: #f5a623;
        }

        .checked > .inner {
          left: calc(100% - (var(--toggle-height) - 2px));
          box-shadow: none;
        }
      </style></label><label class=\\"\\"><input type=\\"checkbox\\"><div class=\\"toggle\\"><span class=\\"inner\\"></span></div><style>
        label {
          -webkit-tap-highlight-color: transparent;
          display: inline-block;
          vertical-align: middle;
          white-space: nowrap;
          user-select: none;
          position: relative;
          cursor: pointer;
          --toggle-font-size: calc(1 * 16px);
          --toggle-height: calc(0.875 * 16px);
          width: calc(1.75 * 16px);
          height: var(--toggle-height);
          padding: calc(0.1875 * 16px) 0 calc(0.1875 * 16px)
            0;
          margin: 0 0 0 0;
        }

        input {
          overflow: hidden;
          visibility: hidden;
          height: 0;
          opacity: 0;
          width: 0;
          position: absolute;
          background-color: transparent;
          z-index: -1;
        }

        .toggle {
          height: var(--toggle-height);
          width: 100%;
          border-radius: var(--toggle-height);
          transition-delay: 0.12s;
          transition-duration: 0.2s;
          transition-property: background, border;
          transition-timing-function: cubic-bezier(0, 0, 0.2, 1);
          position: relative;
          border: 1px solid transparent;
          background-color: #eaeaea;
          padding: 0;
        }

        .inner {
          width: calc(var(--toggle-height) - 2px);
          height: calc(var(--toggle-height) - 2px);
          position: absolute;
          top: 50%;
          transform: translateY(-50%);
          left: 1px;
          box-shadow: rgba(0, 0, 0, 0.2) 0 1px 2px 0, rgba(0, 0, 0, 0.1) 0 1px 3px 0;
          transition: left 280ms cubic-bezier(0, 0, 0.2, 1);
          border-radius: 50%;
          background-color: #fff;
        }

        .disabled {
          border-color: #eaeaea;
          background-color: #fafafa;
        }

        .disabled > .inner {
          background-color: #eaeaea;
        }

        .disabled.checked {
          border-color: #888;
          background-color: #888;
        }

        .checked {
          background-color: #e00;
        }

        .checked > .inner {
          left: calc(100% - (var(--toggle-height) - 2px));
          box-shadow: none;
        }
      </style></label></div>"
`;
